structuralcodes/materials/constitutive_laws.py

```diff
@@ -776,3 +884,54 @@ def set_ultimate_strain(
                 'set_ultimate_strain requires a single value or a tuple \
                 with  two values'
             )
+
+
+CONSTITUTIVE_LAWS: t.Dict[str, ConstitutiveLaw] = {
+    'elastic': Elastic,
+    'elasticplastic': ElasticPlastic,
+    'elasticperfectlyplastic': ElasticPlastic,
+    'bilinearcompression': BilinearCompression,
+    'parabolarectangle': ParabolaRectangle,
+    'popovics': Popovics,
+    'sargin': Sargin,
+}
+
+
+def get_constitutive_laws_list() -> t.List[str]:
+    """Returns a list with valid keywords for constitutive law factory."""
+    return list(CONSTITUTIVE_LAWS.keys())
+
+
+def create_constitutive_law(
+    constitutive_law_name: str, material: Material
+) -> ConstitutiveLaw:
+    """A factory function to create the constitutive law.
+
+    Args:
+        constitutive_law_name (str): a string defining a valid constitutive law
+            type. The available keys can be get with the method
+            `get_constitutive_laws_list`
+        material (Material): The material containing the properties needed for
+            the definition of the constitutive law
+    """
+    law = None
+    const_law = CONSTITUTIVE_LAWS.get(constitutive_law_name.lower())
+    if const_law is not None:
+        method_name = f'__{constitutive_law_name}__'
+        # check if the material object has the special method needed
+        if hasattr(material, method_name):
+            print('The method is there!')
+            method = getattr(material, method_name)
+            if callable(method):
+                kwargs = method()
+                print(kwargs)
```

Comment:
Remember to remove the prints 😄

Reply:
Sorry! I forgot! Done in 16967e2fe4a6290daab9551cb04ae7167cc69c21